src/semver.toit

```diff
@@ -8,188 +8,545 @@ A semantic versioning library.
 See https://semver.org/ for details.
 */
 
-split-semver_ semver/string:
-  plus-index := semver.index-of "+"
-  if plus-index != -1:
-    // Drop the build metadata.
-    // We don't need it for comparison.
-    semver = semver[..plus-index]
-
-  prerelease/string? := null
-  minus-index := semver.index-of "-"
-  if minus-index != -1:
-    prerelease = semver[minus-index + 1..]
-    semver = semver[..minus-index]
-
-  return [semver, prerelease]
-
 /**
-Compares a dotted part of a semver string.
-
-This is used for major.minor.patch, but also for prerelease and build.
-
-Generally, the rules are:
-- If one is an integer and the other is not, the integer is lower.
-- If both are integers compare them.
-- If both are strings compare them.
+Determines if a semantic version string is valid against semver 2.0.0.
+
+This function accepts parameters as defined in README.md:
+- If `$accept-version-core-zero` is true, then 0.0.0 will be accepted for
+the version core.
+- If `$accept-missing-minor` is true, then accepts version numbers without minor
+(and patch), like `1`.
+- If `$accept-missing-patch` is true, then accepts version numbers without
+patch, like `1.2`.
+- If `$accept-v` is true, version numbers are accepted with the preceeding v,
+like `v1.2.1`.
+- If `$accept-leading-zeros` is true, version numbers are accepted that have
+leading zeros in front of them, like `1.02.3`.
```

Comment:
As a user of a package, it isn't necessarily easy to see the README. I can even imagine that the package manager only downloads the `src` directory.

```suggestion
If `$accept-version-core-zero` is true, then 0.0.0 will be accepted for the version core.
If `$accept-missing-minor` is true, then accepts version numbers without minor
  (and patch), like `1`.
If `$accept-missing-patch` is true, then accepts version numbers without patch, like `1.2`.
If `$accept-v` is true, version numbers are accepted with the preceding v, like `v1.2.1`.
If `$accept-leading-zeros` is true, version numbers are accepted that have
  leading zeros in front of them, like `1.02.3`.
```

I removed the lists, but even with lists, you would need to adjust the indentation.